File: ArrayOddSum.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    vector<string> ans;
    while (t--) {
        int n;
        cin >> n;
        int total = 0;
        vector<int> arr(n);
        for (auto &x:arr)
            cin >> x;
        for (auto &x:arr)
            total += x;
        if (total%2!=0)
            ans.push_back("YES");
        else {
            bool even=false, odd=false;
            for (int i=0; i<arr.size(); i++) {
                if (arr[i]%2==0)
                    even = true;
                else if (arr[i]%2!=0)
                    odd = true;
            }
            if (even && odd)
                ans.push_back("YES");
            else 
                ans.push_back("NO");
        }
    }
    for (auto &x:ans)
        cout << x << endl;

    return 0;
}

File: Python/ArrayOddSum.py
from typing import Tuple

from typing_extensions import TypeGuard


def main():
    t = int(input())
    ans = []
    while t:
        n = int(input())
        total = 0
        arr = [int(input()) for i in range(n)]
        if total % 2 != 0:
            ans.append("YES")
        else:
            even = False
            odd = False
            for i in arr:
                if i % 2 == 0:
                    even = True
                else:
                    odd == True
            if even and odd:
                ans.append("YES")
            else:
                ans.append("NO")
        t -= 1
    for i in ans:
        print(i)


if __name__ == '__main__':
    main()
